src/azure-cli/azure/cli/command_modules/cloud/custom.py

```diff
@@ -34,31 +34,22 @@ def show_cloud(cmd, cloud_name=None):
         raise CLIError(e)
 
 
-def _populate_from_metadata_endpoint(cloud, arm_endpoint, session=None):
-    endpoints_in_metadata = ['active_directory_graph_resource_id',
-                             'active_directory_resource_id', 'active_directory']
-    METADATA_ENDPOINT_SUFFIX = '/metadata/endpoints?api-version=2015-01-01'
-    if not arm_endpoint or all([cloud.endpoints.has_endpoint_set(n) for n in endpoints_in_metadata]):  # pylint: disable=use-a-generator
-        return
+def _populate_from_metadata_endpoint(arm_endpoint, session=None):
```

Comment:
Is it really necessary to change the function signature by removing `cloud`?

Reply:
Because now we generate a `Cloud` from none. While calling this func, there's no `cloud` yet so I have nothing to pass in